src/DssSpell.t.base.sol

```diff
@@ -1353,46 +1353,51 @@ contract DssSpellTestBase is Config, DssTest {
             vm.revertTo(snapshot);
         }
         // Check locking and freeing Sky
         {
+            uint256 initialEngineBalance = mkr.balanceOf(p.engine);
             engine.open(0);
             uint256 skyAmt = lockAmt * afterSpell.sky_mkr_rate;
-            assertEq(sky.balanceOf(address(this)), skyAmt, "LockstakeTake/LockAndFreeSky/invalid-initial-balance");
+            assertEq(sky.balanceOf(address(this)), skyAmt, "checkLockstakeIlkIntegration/LockAndFreeSky/invalid-initial-balance");
             sky.approve(address(engine), skyAmt);
             engine.lockSky(address(this), 0, skyAmt, 0);
-            assertEq(mkr.balanceOf(p.engine), lockAmt, "LockstakeTake/LockAndFreeSky/invalid-locked-mkr-balance");
+            assertEq(mkr.balanceOf(p.engine), initialEngineBalance + lockAmt, "checkLockstakeIlkIntegration/LockAndFreeSky/invalid-locked-mkr-balance");
             engine.freeSky(address(this), 0, address(this), skyAmt);
-            uint256 exitFee = skyAmt * p.fee / 10_000;
-            assertEq(sky.balanceOf(address(this)), skyAmt - exitFee, "LockstakeTake/LockAndFreeSky/invalid-unlocked-balance");
+            uint256 exitFee = lockAmt * p.fee / 100_00 * afterSpell.sky_mkr_rate;
+            assertGe(sky.balanceOf(address(this)), skyAmt - exitFee, "checkLockstakeIlkIntegration/LockAndFreeSky/invalid-unlocked-balance");
             vm.revertTo(snapshot);
         }
         // Check drawing and wiping
         {
+            uint256 initialEngineBalance = mkr.balanceOf(p.engine);
             address urn = engine.open(0);
-            assertEq(mkr.balanceOf(address(this)), lockAmt, "LockstakeTake/DrawAndWipe/invalid-initial-balance");
+            assertEq(mkr.balanceOf(address(this)), lockAmt, "checkLockstakeIlkIntegration/DrawAndWipe/invalid-initial-balance");
             mkr.approve(address(engine), lockAmt);
             engine.lock(address(this), 0, lockAmt, 0);
-            assertEq(mkr.balanceOf(p.engine), lockAmt, "LockstakeTake/DrawAndWipe/invalid-locked-mkr-balance");
+            assertEq(mkr.balanceOf(p.engine), initialEngineBalance + lockAmt, "checkLockstakeIlkIntegration/DrawAndWipe/invalid-locked-mkr-balance");
             engine.draw(address(this), 0, address(this), drawAmt);
-            assertEq(usds.balanceOf(address(this)), drawAmt, "LockstakeTake/DrawAndWipe/invalid-usds-balance-after-draw");
+            assertEq(usds.balanceOf(address(this)), drawAmt, "checkLockstakeIlkIntegration/DrawAndWipe/invalid-usds-balance-after-draw");
+            skip(10 days);
+            jug.drip(p.ilk);
             (, uint256 art) = vat.urns(p.ilk, urn);
             (, uint256 rate,,,) = vat.ilks(p.ilk);
             uint256 wipeAmt = _divup(art * rate, RAY);
-            assertGt(wipeAmt, drawAmt, "LockstakeTake/DrawAndWipe/invalid-wipe-after-draw");
+            assertGt(wipeAmt, drawAmt + 1 /* +1 to exclude rounding up */, "checkLockstakeIlkIntegration/DrawAndWipe/invalid-wipe-after-draw");
```

Comment:
I'm not sure if I get what this comment means.

Reply:
Due to the use of `divup` during [engine.draw](https://github.com/makerdao/lockstake/blob/445cd304bab69ff79c15ab947cb48bbe8aadf400/src/LockstakeEngine.sol#L396), the amount of recorded debt can be 1 wei higher than the requested amount. Which resulted in `assertGt` test passing even without warping + dripping (simulating stability fee collection). Both issues were fixed via 40d5caa590c25349244daf7afd8ee0f8f0ae89da

Reply:
Sorry, I had to reopen this.

I understand the intent now, however this only works under the assumption that rate is greater than zero. While this might be a good assumption to make, it would cause the code to fail under that condition.

I'd suggest adding a conditional here and use `assertGe` or `assertEq` in the branch where `rate == 0`.

Reply:
As explained above, `assertEq` will not work even if the rate is 0. That's why `assertGt` was passing without warping and dripping (which is in principle same as having 0 rate)